### DISCUSSION TITLE
How to get table string?

### DISCUSSION THREAD

**Discussion by sadegh-pz** [Textualize/rich#1363] [Q&A]
The documentation only shows how to print tables. But we want to use rich table as string to return from `__repr__` and `__str__` of my custom classes. Is there any way to get rich tables string?

**Comment by willmcgugan** [comment]
I wouldn't recommend returning a string with formatting or ansi codes from repr and str. They may end up formatted badly depending on the context.

However if you really want to do it, you can [capture](https://rich.readthedocs.io/en/latest/console.html#capturing-output) the output.
